### PR TITLE
Fix widget locking world for longer than necessary

### PR DESCRIPTION
**Pull request by mamiksik**
`rtt::world::World::instance()` returns a guard lock, which gets released
at the end of the scope (used to be once the paintEvent returned). This
causes STP to be delayed if `Input::getDrawings()` vector is large (over
3000). Since the lock release is postponed until the method finishes painting all data in the vector (despite not needing the world anymore).

Releasing the lock imminently after the field finished painting mitigates
this issue. As far as I can tell, that is the intended behavior since
`Input::getDrawings()` holds the separate lock. Thus, it does not require
the world to be locked during drawing.

### Comments for the reviewer
- 

###### Code Quality
- [x] Is the code is understandable and easy to read
- [x] Changes to the code comply with set clang-format rules
- [x] No use of manual memory control (e.g new/malloc/colloc etc)
- [x] Are (only) smart pointers used?

###### Testing
- [x] All tests are passing.
- [x] I _added new / changed existing_ tests to reflect code changes (state why not otherwise!)
- [x] I tested my changes manually (Describe how, to what extent etc.)

###### Commit Messages
- [x] Commit message is saying what has been changed, **why** it was changed? Remember other developers might not know
  what the problem you are fixing was. Note also negative _decision_ (e.g., why did you not do particular thing)
  **TLDR: Commit message are comprehensive**
- [x] Commit messages follows the rules of https://chris.beams.io/posts/git-commit/

### CODE REVIEW

**Comment by mamiksik** [comment]
Update, overfilling drawings causes deadlock I think.

File: src/interface/widgets/widget.cpp [hunk 1]
```diff
@@ -21,30 +21,32 @@ void Visualizer::paintEvent(QPaintEvent *event) {
...
-    }
+    {
+        // worldPtr is scoped in order to release the lock soon as possible
+        auto const &[_, worldPtr] = rtt::world::World::instance();
+        auto const &world = worldPtr->getWorld();
+        auto const &field = worldPtr->getField();
 
-    auto s = QString::fromStdString("We have " + std::to_string(world->getUs().size()) + " robots");
-    painter.drawText(24, 48, s.fromStdString("We have " + std::to_string(world->getUs().size()) + " robots"));
+        if (!world.has_value() || !world.value()->weHaveRobots()) {
```
Comment (alexander-de-ranitz) [inline src/interface/widgets/widget.cpp]: I've always been bothered by the fact that we say we done have a world if we don't have any robots. Could we change this to be shown only when there is no world? If we have no robots, we can still draw the field (and enemy robots)

**Comment by Florisheinen1** [comment]
I approve of the changes made, but I also think it is nice to implement the requested change Alex suggested.

File: src/interface/widgets/widget.cpp (hunk 1 above)
Comment (mamiksik) [reply src/interface/widgets/widget.cpp]: ok I will have look into that

**Comment by mamiksik** [comment]
@alexander-de-ranitz @Florisheinen1 Fixed

**Review by Florisheinen1 — APPROVED** [review]
LGTM

**Pull request merged**